trustwallet/wallet-core
# Disable generating Extended Keys for coins TWHDVersionNone

## Description

We are not able to generate valid extendedPublicKey for coins w/o valid Key Version like Ethereum, Ethereum Classic. Wallet Core set extendedPublicKey to "" in all cases except this one:

 `let key = StoredKey.importHDWallet(mnemonic: words, name: "name", password: password, coin: .ethereum)!`
       
If such coin is provided directly to `StoredKey.importHDWallet` it will have extendedPublicKey (invalid) generated.

![Zrzut ekranu 2019-08-9 o 14 36 11](https://user-images.githubusercontent.com/96502/62779543-ca73a680-bab3-11e9-8147-ab3524631b46.png)

Adding Ethereum later will not cause such case because it was checked in StoredKey.cpp.

I think that it will be better to move this check to getExtendedPublicKey and getExtendedPrivateKey methods.

## Testing instructions

* Run tests 
I've added check with .ethereum added directly in importHDWallet and .ethereumClassic via getAccount - for booth extendedPublicKey should be empty.

## Types of changes



* Bug fix (non-breaking change which fixes an issue)





## Checklist
